modules/webgl/src/context/context.js

```diff
@@ -202,12 +196,17 @@ function getVersion(gl) {
 }
 
 // use devicePixelRatio to set canvas width and height
-function setDevicePixelRatio(gl, devicePixelRatio, options) {
+function setDevicePixelRatio(gl, devicePixelRatio) {
   // NOTE: if options.width and options.height not used remove in v8
-  const clientWidth =
-    'width' in options ? options.width : gl.canvas.clientWidth || gl.canvas.width || 1;
-  const clientHeight =
-    'height' in options ? options.height : gl.canvas.clientHeight || gl.canvas.height || 1;
+  let {clientWidth, clientHeight} = gl.canvas;
+
+  if (!clientWidth || !clientHeight) {
+    log.warn('Canvas clientWidht orclientHeight is 0, using devicePixelRatio of 1.0')();
```

Comment:
Remove the warning? It's most likely intentional if the canvas is detached from the document, or at least very obvious to the user.

Reply:
This helps isolate the problem, when it occurs,  I can reduce the string to fewer words, if the bundle size is a concern.

Reply:
Maybe use `log.log` with a lower priority?

Reply:
Done.